### Steam Games Exploratory data analysis
**Author:** Jan Chudy

## Introduction
This end-to-end data science project is the final assignment for a Data Science course at Milwaukee School of Engineering. In this project, we will explore real world data which contains information about video games published on Steam - Digital distribution platform developed by Valve Corporation. 

We will go through the basic process of data science and explore this particular data in the following sections.

### References:
* <a href="https://data.world/craigkelly/steam-game-data">Steam Game Data [dataset]</a> by Craig Kelly
* <a href="https://github.com/jayurbain/DataScienceIntro">Introduction to Data Science</a> by Jay Urbrain, PhD.
* <a href="https://www.analyticsvidhya.com/blog/2016/01/guide-data-exploration/">A Comprehensive Guide to Data Exploration</a> by Sunil Ray


## Hypothesis
First of all, we need to get our hypotheses ready. Here are some questions we will try to answer and problems we will try to solve:
* **Do Free-To-Play games have more DLCs?**
    * Free games, in order to make profit, could have more paid DLCs than games that are not free-to-play. Of course, there are other methods (e.g. Microtransactions) which could be more preferable among free-to-play games.
    * After a brief search on Steam, we can see that paid DLCs are pretty common among the bigger free-to-play games (extensions packs, collector packs...).
    * **Hypothesis 1:** Free-to-Play games have on average more DLCs than paid games.
* **Examine increase in games that support VR according to time**
    * We could see a rise in popularity of VR games in recent years. How moch games with VR support have been released? Can we see a significant increase in this type of games?
    * **Hypothesis 2:** Each year the amount of games with VR support being released increases.
* **Examine game rating averages in time.**
    * **Hypothesis 3:** Overall rating averages are decreasing over time.
* **What features contribute to a game's rating and can we predict it?**

Besides these questions, we will also try to take as much observations as we can to understand this data better.

## Dataset
For this Data Science project, we will use a dataset created by Craig Kelly, that contains 13357 steam games (unfortunately with some duplicates).

### Information
**Name**: Steam Game Data<br>
**Author**: <a href="https://data.world/craigkelly">Craig Kelly</a><br>
**Data Source**: Combination of data from publicly available Steam API's and steamspy.com<br>
**Links**:
* Data set available on <a href="https://data.world/craigkelly/steam-game-data">data.world</a> (with details)
* Full details on <a href="https://github.com/CraigKelly/steam-data">GitHub</a> 

### Attributes
* **QueryID** - (Integer) The original ID
* **ResponseID** - (Integer) The ID returned in the Steam response (should equal QueryID)
* **QueryName** - (Text) The original name
* **ResponseName** - (Text) The name returned in the Steam response (should equal QueryName)
* **ReleaseDate** - (Text) Appears to the be the initial release date for the game
* **RequiredAge** - (Integer) required age for the game
* **DemoCount** - (TextualCount) number of demos
* **DeveloperCount** - (TextualCount) number of developers
* **DLCCount** - (TextualCount) number of DLCs
* **Metacritic** - (Integer) numeric score from metacritic
* **MovieCount** - (TextualCount) number of videos on Store Page
* **PackageCount** - (TextualCount) number of bundles the game occurs in
* **RecommendationCount** - (Integer) number of recomendations
* **PublisherCount** - (TextualCount) number of publishers
* **ScreenshotCount** - (TextualCount) number of screenshots on Store Page
* **AchievementCount** - (Integer) number of achievements
* **AchievementHighlightedCount** - (TextualCount) number of highlighted achievements
* **ControllerSupport** - (Boolean) True if the game has full controller support
* **IsFree** - (Boolean) True if the game is free-to-play
* **FreeVerAvail** - (Boolean) True if is_free_license is True in package_groups list
* **PurchaseAvail** - (Boolean) True if price_in_cents_with_discount is greater than 0 in package_groups list
* **SubscriptionAvail** - (Boolean) True if is_recurring_subscription is True in package_groups
* **PlatformWindows** - (Boolean) True if Windows platform is supported
* **PlatformLinux** - (Boolean) True if Linux platform is supported
* **PlatformMac** - (Boolean) True if Mac platform is supported
* **PCReqsHaveMin** - (Boolean) True if PC minimal requirements are available
* **PCReqsHaveRec** - (Boolean) True if PC recommended requirements are available
* **LinuxReqsHaveMin** - (Boolean) True if Linux minimal requirements are available
* **LinuxReqsHaveRec** - (Boolean) True if Linux recommended requirements are available
* **MacReqsHaveMin** - (Boolean) True if Mac minimal requirements are available
* **MacReqsHaveRec** - (Boolean) True if Mac recommended requirements are available
* **CategorySinglePlayer** - (Boolean) True if category is Singleplayer
* **CategoryMultiplayer** - (Boolean) True if category is Cross-platform multiplayer/Local multi-player/Multi-player/Online multi-player/Shared screen/Split screen
* **CategoryCoop** - (Boolean) True if category is Co-op/Local co-op/Online co-op
* **CategoryMMO** - (Boolean) True if category is MMO
* **CategoryInAppPurchase** - (Boolean) True if game has in-app purchases
* **CategoryIncludeSrcSDK** - (Boolean) True if game includes source sdk
* **CategoryIncludeLevelEditor** - (Boolean) True if game includes level editor
* **CategoryVRSupport** - (Boolean) True if game has VR support
* **GenreIsNonGame** - (Boolean) True if genre is Utilities/Design & illustration/Animation & modeling/Software training/Education/Audio production/Video production/Web publishing/Photo editing/Accounting
* **GenreIsIndie** - (Boolean) True if genre is indie
* **GenreIsAction** - (Boolean) True if genre is action
* **GenreIsAdventure** - (Boolean) True if genre is adventure
* **GenreIsCasual** - (Boolean) True if genre is casual
* **GenreIsStrategy** - (Boolean) True if genre is strategy
* **GenreIsRPG** - (Boolean) True if genre is rpg
* **GenreIsSimulation** - (Boolean) True if genre is simulation
* **GenreIsEarlyAccess** - (Boolean) True if genre is early access
* **GenreIsFreeToPlay** - (Boolean) True if genre is free to play
* **GenreIsSports** - (Boolean) True if genre is sports
* **GenreIsRacing** - (Boolean) True if genre is racing
* **GenreIsMassivelyMultiplayer** - (Boolean) True if genre is massively multiplayer
* **PriceCurrency** - (Text) Currency of the price
* **PriceInitial** - (Float) Initial price of the game
* **PriceFinal** - (Float) Final price of the game
* **SteamSpyOwners** - [steamspy.com] total owners, which includes free weekend trials and other possibly spurious numbers
* **SteamSpyOwnersVariance** - [teamspy.com] total owners, which includes free weekend trials and other possibly spurious numbers. Note that this is not technically variance: according to steamspy.com, "the real number... lies somewhere on... (value +/- variance)"
* **SteamSpyPlayersEstimate** - [steamspy.com] best estimate of total number of people who have played the game since March 2009
* **SteamSpyPlayersVariance** - [steamspy.com] errors bounds on SteamSpyPlayersEstimate. Note that this is not technically variance: according to steamspy.com, "the real number... lies somewhere on... (value +/- variance)"
* **SupportEmail** - (Textual) Support email address
* **SupportURL** - (Textual) Support URL
* **AboutText** - (Textual) About text
* **Background** - (Textual) Background of the game
* **ShortDescrip** - (Textual) Short description of the game
* **DetailedDescrip** - (Textual) Detailed description of the game
* **DRMNotice** - (Textual) Digital rights management notice
* **ExtUserAcctNotice** - (Textual) ext_user_account_notice in JSON
* **HeaderImage** - (Textual) Header image
* **LegalNotice** - (Textual) Legal notice
* **Reviews** - (Textual) Reviews
* **SupportedLanguages** - (Textual) Supported langiages in the game
* **Website** - (Textual) Game website
* **PCMinReqsText** - (Textual) PC minimal requirements
* **PCRecReqsText** - (Textual) PC recommended requirements
* **LinuxMinReqsText** - (Textual) Linux minimal requirements
* **LinuxRecReqsText** - (Textual) Linux recommended requirements
* **MacMinReqsText** - (Textual) Mac minimal requirements
* **MacRecReqsText** - (Textual) Mac recommended requirements

## Data preprocessing
In this section we will look on the actual data closely and preprocess the data for our analysis and modeling. 

Let's include some libraries we will need.

In [1]:
%matplotlib inline 

import pandas as pd #import pandas
import numpy as np #import numpy
import matplotlib.pyplot as plt # set up plotting under plt

# our helper functions from utils.py
exec(compile(open("utils/utils.py", "rb").read(), "utils/utils.py", 'exec'))

#sets up pandas table display
pd.set_option('display.max_columns', 100)
pd.set_option('display.notebook_repr_html', True)

We will read the dataset into pandas for easy manipulation. 

In [2]:
df = pd.read_csv("data/games-features.csv") # read the data into panda's data frame
print('The shape of the dataset is {}.'.format(df.shape)) # print out the shape of the dataset
df.head(10) # show first 10 records

The shape of the dataset is (13357, 78).


QueryID  ResponseID                       QueryName  \
0       10          10                  Counter-Strike   
1       20          20           Team Fortress Classic   
2       30          30                   Day of Defeat   
3       40          40              Deathmatch Classic   
4       50          50       Half-Life: Opposing Force   
5       60          60                        Ricochet   
6       70          70                       Half-Life   
7       80          80  Counter-Strike: Condition Zero   
8       80          80  Counter-Strike: Condition Zero   
9      130         130           Half-Life: Blue Shift   

                     ResponseName ReleaseDate  RequiredAge  DemoCount  \
0                  Counter-Strike  Nov 1 2000            0          0   
1           Team Fortress Classic  Apr 1 1999            0          0   
2                   Day of Defeat  May 1 2003            0          0   
3              Deathmatch Classic  Jun 1 2001            0          0   
4       Half-Life: Opposing Force  Nov 1 1999            0          0   
5                        Ricochet  Nov 1 2000            0          0   
6                       Half-Life  Nov 8 1998            0          0   
7  Counter-Strike: Condition Zero  Mar 1 2004            0          0   
8  Counter-Strike: Condition Zero  Mar 1 2004            0          0   
9           Half-Life: Blue Shift  Jun 1 2001            0          0   

   DeveloperCount  DLCCount  Metacritic  MovieCount  PackageCount  \
0               1         0          88           0             1   
1               1         0           0           0             1   
2               1         0          79           0             1   
3               1         0           0           0             1   
4               1         0           0           0             1   
5               1         0           0           0             1   
6               1         1          96           0             1   
7               1         0          65           0             1   
8               1         0          65           0             1   
9               1         0          71           0             1   

   RecommendationCount  PublisherCount  ScreenshotCount  SteamSpyOwners  \
0                68991               1               13        13033334   
1                 2439               1                5         5399140   
2                 2319               1                5         7621102   
3                  888               1                4         7498965   
4                 2934               1                5         5399499   
5                 1965               1                4         7488563   
6                12486               1               11         5927504   
7                 7067               1                8        10489261   
8                 7067               1                8        10489261   
9                 2219               1                5         5368650   

   SteamSpyOwnersVariance  SteamSpyPlayersEstimate  SteamSpyPlayersVariance  \
0                   92789                  9140731                    78136   
1                   60368                   753627                    22699   
2                   71499                  1709740                    34145   
3                   70936                   789676                    23235   
4                   60370                   705741                    21968   
5                   70888                   843481                    24011   
6                   63206                  2268772                    39303   
7                   83543                  4127910                    52878   
8                   83543                  4127910                    52878   
9                   60199                   662517                    21286   

   AchievementCount  AchievementHighlightedCount  ControllerSupport  IsFree  \
0                 0      

Now let's go through all attributes and determine which attributes we will need for our analysis. We can list all 78 atributes:

In [3]:
columns = list(df.columns)
print(columns)

['QueryID', 'ResponseID', 'QueryName', 'ResponseName', 'ReleaseDate', 'RequiredAge', 'DemoCount', 'DeveloperCount', 'DLCCount', 'Metacritic', 'MovieCount', 'PackageCount', 'RecommendationCount', 'PublisherCount', 'ScreenshotCount', 'SteamSpyOwners', 'SteamSpyOwnersVariance', 'SteamSpyPlayersEstimate', 'SteamSpyPlayersVariance', 'AchievementCount', 'AchievementHighlightedCount', 'ControllerSupport', 'IsFree', 'FreeVerAvail', 'PurchaseAvail', 'SubscriptionAvail', 'PlatformWindows', 'PlatformLinux', 'PlatformMac', 'PCReqsHaveMin', 'PCReqsHaveRec', 'LinuxReqsHaveMin', 'LinuxReqsHaveRec', 'MacReqsHaveMin', 'MacReqsHaveRec', 'CategorySinglePlayer', 'CategoryMultiplayer', 'CategoryCoop', 'CategoryMMO', 'CategoryInAppPurchase', 'CategoryIncludeSrcSDK', 'CategoryIncludeLevelEditor', 'CategoryVRSupport', 'GenreIsNonGame', 'GenreIsIndie', 'GenreIsAction', 'GenreIsAdventure', 'GenreIsCasual', 'GenreIsStrategy', 'GenreIsRPG', 'GenreIsSimulation', 'GenreIsEarlyAccess', 'GenreIsFreeToPlay', 'GenreIsS

From the first 4 columns ('QueryID', 'ResponseID', 'QueryName', 'ResponseName'), we are really interested only in the response names. After a quick examination we would be able to see hat there are som inconsistencies between 'QueryID' and 'ResponseID'. There are a lot more entries with 'QueryName' and 'ResponseName' mismatch. The reason for this is most likely that the Response Names are more specific and that's what we are interested - the actual names of the games. We can drop the first three columns and rather look for duplicates in the entries (there should be a lot of them).

In [4]:
# only for perspective
print('There are {} entries, where QuerryID != ResponseID.'.format(len(df[df.QueryID != df.ResponseID])))
print('There are {} entries, where QueryName != ResponseName.'.format(len(df[df.QueryName != df.ResponseName])))

There are 98 entries, where QuerryID != ResponseID.
There are 2499 entries, where QueryName != ResponseName.


In [5]:
# drop first three columns
df.drop(['QueryID', 'ResponseID', 'QueryName'], axis=1, inplace=True)
df.shape

(13357, 75)

In [6]:
# count duplicates
len(df[df.duplicated(subset=['ResponseName'], keep=False)])

239

We can see there are 239 entries that are duplicates (the ResponseName has more than one appearance). We might want to keep the first appearance of the game. But if we go through the duplicites we'll find out that the first appierance is sometimes invalid. We'll have to use a bit more sophisticated duplication deletion. If we examine the duplicates, we can see that some of them does have missing values in attributes 'SteamSpyOwners', 'SteamSpyOwnersVariance', 'SteamSpyPlayersEstimate' and 'SteamSpyPlayersVariance' or the values are significantly lower than for the other duplicates of the same game. Based on this observation, we can decide which entry do we leave in the dataset.

In [7]:
# delete invalid duplicates from the dataset
df = delete_duplicates(df) # function from our helper functions
df.shape

(13199, 75)

In [8]:
# count duplicates again
print('There are now {} duplicates.'.format(len(df[df.duplicated(subset=['ResponseName'], keep=False)])))

There are now 0 duplicates.


Now that we dropped all duplicates, let's get rid of all entries that are not games. We can use 'GenreIsNonGame' column for this (as a mask for dropping entries). After that, we can also drop the column.

In [9]:
print('There are {} non-game entreis in the dataset.'.format(len(df[df.GenreIsNonGame == True])))
df.drop(df[df.GenreIsNonGame == True].index, axis=0, inplace=True) # drop all non-Game entries
print('Now there are are {} non-game entreis in the dataset.'.format(len(df[df.GenreIsNonGame == True])))
df.drop(['GenreIsNonGame'], axis=1, inplace=True) # drop the column
df.shape # check how the shape of the dataset changed

There are 318 non-game entreis in the dataset.
Now there are are 0 non-game entreis in the dataset.


(12881, 74)

Apparently, there were some non-game entries that has 'GenreIsNonGame' == False. We will have to remove them manually, but we most likely won't be able to clear all the data.

In [10]:
# Manual cleaning of the data based on manual search on Steam
# videos or movies:
df.drop(df[df.ResponseName.str.contains('From Bedrooms to Billions')].index, axis=0, inplace=True)
df.drop(df[df.ResponseName.str.contains('PAYDAY: The Web Series')].index, axis=0, inplace=True)
df.drop(df[df.ResponseName.str.contains('The Making of Grim Fandango Remastered')].index, axis=0, inplace=True)
df.drop(df[df.ResponseName.str.contains('Mortal Kombat: Legacy')].index, axis=0, inplace=True)
df.drop(df[df.ResponseName.str.contains('Mortal Kombat: Legacy II')].index, axis=0, inplace=True)
df.drop(df[df.ResponseName.str.contains('Double Fine Adventure')].index, axis=0, inplace=True)
df.drop(df[df.ResponseName.str.contains('Naruto Shippuden Uncut')].index, axis=0, inplace=True)
df.drop(df[df.ResponseName.str.contains('Complete Figure Drawing Course')].index, axis=0, inplace=True)
df.drop(df[df.ResponseName.str.contains('Kalen Chock Presents')].index, axis=0, inplace=True)
df.drop(df[df.ResponseName.str.contains('Now You See Me')].index, axis=0, inplace=True)
df.drop(df[df.ResponseName.str.contains('Robotpencil Presents')].index, axis=0, inplace=True)
df.drop(df[df.ResponseName.str.contains('Raiders! : The Story of the Greatest Fan Film')].index, axis=0, inplace=True)
df.drop(df[df.ResponseName.str.contains('Man vs Snake: The Long and Twisted Tale')].index, axis=0, inplace=True)
df.drop(df[df.ResponseName.str.contains('CS:GO Player Profiles')].index, axis=0, inplace=True)
df.drop(df[df.ResponseName.str.contains('Lets Play:')].index, axis=0, inplace=True)
df.drop(df[df.ResponseName.str.contains('Dota 2 Player Profiles')].index, axis=0, inplace=True)
df.drop(df[df.ResponseName.str.contains('Ghost in the Shell')].index, axis=0, inplace=True)
df.drop(df[df.ResponseName.str.contains('Meet the Blacks')].index, axis=0, inplace=True)
df.drop(df[df.ResponseName.str.contains('Kindred Spirits on the Roof Drama')].index, axis=0, inplace=True)

df.shape

(12260, 74)

Now we can proceed with other attributes.

We won need 'AchievementHighlightedCount' attribute. This attribute contains a number of higlighted achievements, which is a number from 0 - 10 depending on the number of achievements. If a game has 10 or more achievements, the 'AchievementHighlightedCount' is equal to 10. Otherwise it is equal to 'AchievementCount'. We can drop this column.

In [11]:
df.drop(['AchievementHighlightedCount'], axis=1, inplace=True)

We also wont use minimal and maximal recommendations for these games. Also a big portion of the dataset miss these values. We're dropping 'PCReqsHaveMin', 'PCReqsHaveRec', 'LinuxReqsHaveMin', 'LinuxReqsHaveRec', 'MacReqsHaveMin', 'MacReqsHaveRec', 'PCMinReqsText', 'PCRecReqsText', 'LinuxMinReqsText', 'LinuxRecReqsText', 'MacMinReqsText' and 'MacRecReqsText'.

In [12]:
columns_to_drop = ['PCReqsHaveMin', 'PCReqsHaveRec', 'LinuxReqsHaveMin', 'LinuxReqsHaveRec', 'MacReqsHaveMin',
                   'MacReqsHaveRec', 'PCMinReqsText', 'PCRecReqsText', 'LinuxMinReqsText', 'LinuxRecReqsText',
                   'MacMinReqsText', 'MacRecReqsText']
df.drop(columns_to_drop, axis=1, inplace=True)

There are more columns we won't need. For example all the support contact information, legal notices and a website. So we can drop 'SupportEmail', 'SupportURL', 'DRMNotice', 'ExtUserAcctNotice', 'LegalNotice' and 'Website'.

In [13]:
columns_to_drop = ['SupportEmail', 'SupportURL', 'DRMNotice', 'ExtUserAcctNotice', 'LegalNotice', 'Website']
df.drop(columns_to_drop, axis=1, inplace=True)

Also some descriptions aren't very helpful for our purposes. Though, it might be useful to have the information if certain games have or does not have these descriptions. The same goes with images (background and header). Let's have a look at the 'AboutText', 'ShortDescrip' and 'DetailedDescrip'. The 'ShortDescrip' is a short description of the game that can be found right under the header image on the Store Page of the game. Some games have it, some don't and sometimes the short description is a part of 'AboutText'. We will change this into a Boolean attribute with True value if the game has a short description and False if it has no short description.

In [14]:
print('{} of our entries have no short description.'.format(len(df[df.ShortDescrip == ' '])))
df.replace({'ShortDescrip': {' ': ''}}, inplace=True) # replace with empty string in order to get False when retyped
df.ShortDescrip = df.ShortDescrip.astype(bool) # retype to Boolean

1728 of our entries have no short description.


In [15]:
# check the result
print(df.ShortDescrip.dtype)
len(df[df.ShortDescrip == False])

bool


1728

Now the 'AboutText' and 'DetailedDescrip'. 'AboutText' is a text on the game's Store Page with title "ABOUT THIS GAME", so it make sense for us to note if a game has this text available. On the other hand 'DetailedDescrip' is nowhere to be found on the Store Page. If we look at the dataset, we can observe, that a big portion of the 'DetailedDescrip' is identical to 'AboutText' and others seem to be some Update descriptions or are simply missing. We will drop this column and retype 'AboutText' into boolean.

In [16]:
p = len(df[df.AboutText == df.DetailedDescrip]) / len(df) * 100
print("{:0.2f}% of the 'DetailedDescrip' is identical to 'AboutText'.".format(p))

df.drop(['DetailedDescrip'], axis=1, inplace=True) # drop 'DetailedDescrip'

90.50% of the 'DetailedDescrip' is identical to 'AboutText'.


Now we only have to take care of the 'AboutText'. Let's have a look at it.

In [17]:
print('{} of our entries have no "about the game" text.'.format(len(df[df.AboutText == ' '])))

45 of our entries have no "about the game" text.


Note that some of these entries are missing the 'AboutText' value because there is no text in the "ABOUT THIS GAME" section on Store Page of the game. If we search these games on steam, we'll find out that some of the games have an image version of the text (with other gaphics) instead of text. Here are some exception we'll fix manually:

In [18]:
df.loc[df[df.ResponseName == 'iREC'].index, 'AboutText'] = 'True'
df.loc[df[df.ResponseName == 'Crypt of the NecroDancer'].index, 'AboutText'] = 'True'
df.loc[df[df.ResponseName == 'Monochroma'].index, 'AboutText'] = 'True'
df.loc[df[df.ResponseName == 'No Turning Back: The Pixel Art Action-Adventure Roguelike'].index, 'AboutText'] = 'True'
df.loc[df[df.ResponseName == 'Crazy Pixel Streaker'].index, 'AboutText'] = 'True'
df.loc[df[df.ResponseName == 'One Thousand Lies'].index, 'AboutText'] = 'True'
df.loc[df[df.ResponseName == 'Azurea Juncture'].index, 'AboutText'] = 'True'

# NOTE: value 'True' can be any non-empty string at this point...

print('Now, {} of our entries have no "about the game" text.'.format(len(df[df.AboutText == ' '])))

Now, 38 of our entries have no "about the game" text.


We can now store only a boolean value if a game has "About the game" section or not.

In [19]:
df.replace({'AboutText': {' ': ''}}, inplace=True) # replace with empty string in order to get False when retyped
df.AboutText = df.AboutText.astype(bool) # retype to Boolean

We will change 'Background' and 'HeaderImage' in a similar fashion.

In [20]:
print("{} of games don't have a background image and {} of games don't have a header image."\
      .format(len(df[df.Background==' ']),len(df[df.HeaderImage==' '])))

84 of games don't have a background image and 0 of games don't have a header image.


Apparently, there are no entries that are missing 'HeaderImage' what makes this attribute useless for our purposes and we can drop it.

In [21]:
df.drop(['HeaderImage'], axis=1, inplace=True) # drop 'HeaderImage'

# retype 'Background'
df.replace({'Background': {' ': ''}}, inplace=True) # replace with empty string in order to get False when retyped
df.Background = df.Background.astype(bool) # retype to Boolean

In [23]:
columns = list(df.columns)
print(columns)

['ResponseName', 'ReleaseDate', 'RequiredAge', 'DemoCount', 'DeveloperCount', 'DLCCount', 'Metacritic', 'MovieCount', 'PackageCount', 'RecommendationCount', 'PublisherCount', 'ScreenshotCount', 'SteamSpyOwners', 'SteamSpyOwnersVariance', 'SteamSpyPlayersEstimate', 'SteamSpyPlayersVariance', 'AchievementCount', 'ControllerSupport', 'IsFree', 'FreeVerAvail', 'PurchaseAvail', 'SubscriptionAvail', 'PlatformWindows', 'PlatformLinux', 'PlatformMac', 'CategorySinglePlayer', 'CategoryMultiplayer', 'CategoryCoop', 'CategoryMMO', 'CategoryInAppPurchase', 'CategoryIncludeSrcSDK', 'CategoryIncludeLevelEditor', 'CategoryVRSupport', 'GenreIsIndie', 'GenreIsAction', 'GenreIsAdventure', 'GenreIsCasual', 'GenreIsStrategy', 'GenreIsRPG', 'GenreIsSimulation', 'GenreIsEarlyAccess', 'GenreIsFreeToPlay', 'GenreIsSports', 'GenreIsRacing', 'GenreIsMassivelyMultiplayer', 'PriceCurrency', 'PriceInitial', 'PriceFinal', 'AboutText', 'Background', 'ShortDescrip', 'Reviews', 'SupportedLanguages']


Let's look at the 'Reviews' attribute. Apaprently, majority of the dataset miss this values. There might have been a problem with the extraction of the data during the creation of the dataset. Nevertheless, the text form of reviews wouldn't be very helpful for our purposes. We assume, that 'RecommendationCount' is the number of Reviews (or positive Reviews only?). Therefore, we'll drop this column.

In [30]:
p = len(df[df.Reviews == ' ']) / len(df) * 100 
print("{:0.2f}% of the 'Reviews' values are empty.".format(p))

df.drop(['Reviews'], axis=1, inplace=True) # drop 'Reviews'

73.62% of the 'Reviews' values are empty.


In [33]:
len(df.SupportedLanguages.unique())

2321

In [34]:
len(df[df.SupportedLanguages == ' '])

20

In [ ]:
TODO: LOOK AT THE SUPPORTEDLANGUAGES

In [38]:
#df.SupportedLanguages.unique()

## Data analysis and visualization

## Data modeling and prediction

## Analysis results